sdk/cosmosdb/cosmos/src/ClientContext.ts

```diff
@@ -333,6 +333,7 @@ export class ClientContext {
         options,
         plugins: this.cosmosClientOptions.plugins,
         partitionKey,
+        pipeline: this.pipeline,
```

Comment:
Is pipepline is MUST always can it be enforced? (ex: like through constructor) etc...

This was discovered is by accident may be this change will make it complete. But future changes might accidentally introduce it (ex: new API).

Reply:
I agree that if it's an error to omit this it should be marked required and enforced.

Reply:
Totally agree with the suggestion, unfortunately pipelines and some other properties are not mandatory, and they depend upon context. i.e. pipeline is used only in case of aadCredentials, whereas in case of 'key' it's not. One middle ground I have been able to find is to move assembly of all the ClientContext dependent properties to a separate method which is used by all APIs.